### PR TITLE
Testserver client javadocs

### PR DESCRIPTION
**Pull request by olensmar**

### CODE REVIEW

File: readyapi4j-core/src/main/java/com/smartbear/readyapi4j/result/AbstractTestStepResult.java [hunk 1]
```diff
@@ -10,6 +10,10 @@
 import java.util.Collections;
 import java.util.List;
 
+/**
+ * Utility class for TestStepResultReport based TestStepResult implementations
```
Comment (prakashmrec) [inline readyapi4j-core/src/main/java/com/smartbear/readyapi4j/result/AbstractTestStepResult.java]: I don't think we should call it utility class. Generally a utility class is with static methods and without state, but this is an abstract class.

File: readyapi4j-testserver-client/src/main/java/com/smartbear/readyapi4j/testserver/RepositoryProjectExecutionRequest.java [hunk 2]
```diff
@@ -6,6 +6,10 @@
 import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * Request object used when executing tests in server repository project
```
Comment (prakashmrec) [inline readyapi4j-testserver-client/src/main/java/com/smartbear/readyapi4j/testserver/RepositoryProjectExecutionRequest.java]: or may be: 'Request object used to execute tests in/on a project available in a repository on TestServer'.

File: readyapi4j-testserver-client/src/main/java/com/smartbear/readyapi4j/testserver/execution/ProjectExecutionRequest.java [hunk 3]
```diff
@@ -100,12 +131,20 @@ private CustomProperties getCustomPropertiesForTargetName(String targetName) {
...
             return projectExecutionRequest;
         }
 
-        public static Builder newInstance() {
-            return new Builder();
+        /**
+         * @return a new instance of this builder for the specified project
+         */
+
+        public static Builder forProjectFile(File projectFile ) {
```
Comment (prakashmrec) [inline readyapi4j-testserver-client/src/main/java/com/smartbear/readyapi4j/testserver/execution/ProjectExecutionRequest.java]: :+1:

**Comment by prakashmrec** [comment]
Looks good, I have added couple of minor comments.

**Comment by olensmar** [comment]
thanks - will fix!

**Comment by olensmar** [comment]
@prakashmrec i've fixed your comments and also improved the RepositoryProjectExecutionRequest class somewhat

**Comment by prakashmrec** [comment]
:+1:  looks good to me, @spindelmanne please have a look and merge it, if it looks good to you.

**Comment by spindelmanne** [comment]
Nice! Merging

**Pull request merged**